math/check_amicable_pair.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2020 @author iamnambiar
+ * 
+ * @file
+ * A C++ Program to check whether a pair of number is amicable pair or not.
+ * 
+ * \brief
+ * Amicable Pair are two different numbers such that sum of the proper divisor 
+ * of each number is equal to the other number.
+ *  
+ * Wikipedia link for Reference: https://en.wikipedia.org/wiki/Amicable_numbers 
+ */
+#include <iostream>
+#include <cassert>
+
+/**
+ * Function to calculate the sum of all the proper divisor 
+ * of an integer.
+ * @param num First number.
+ * @return Sum of the proper divisor of the number.
+ */
+int sum_of_divisor(int num) {
+    // Variable to store the sum of all proper divisors.
+    int sum = 0;
+    // Below loop condition helps to reduce Time complexity by a factor of square root of the number.
+    for (int div = 2; div * div <= num; ++div) {
+        // Check 'div' is divisor of 'num'.
+        if (num % div == 0) {
+            // If both divisor are same, add once to 'sum'
+            if (div == (num / div)) {
+                sum += div;
+            } else {
+                // If both divisor are not the same, add both to 'sum'.
+                sum += (div + (num / div));
+            }
+        }
+    }
+    return sum + 1;
+}
+
+/**
+ * Function to check whether the pair is amicable or not.
+ * @param x First number.
+ * @param y Second number.
+ * @return `true` if the pair is amicable
+ * @return `false` if the pair is not amicable
+ * is not amicable.
+ */
+bool is_amicable(int x, int y) {
+    if (sum_of_divisor(x) != y) {
+        return false;
+    }
+    return (sum_of_divisor(y) == x);
```

Comment:
👍 I personally tested on all 46 numbers from the [OEIS list A259180](https://oeis.org/A259180/b259180.txt)

Reply:
Cool! I have updated the code. Thanks for the review.